biome.json

```diff
@@ -0,0 +1,130 @@
+{
+  "$schema": "https://biomejs.dev/schemas/1.9.4/schema.json",
+  "vcs": {
+    "enabled": true,
+    "clientKind": "git",
+    "useIgnoreFile": true
+  },
+  "files": {
+    "maxSize": 10000000
+  },
+  "formatter": {
+    "indentStyle": "space",
+    "lineWidth": 120,
+    "ignore": ["**/node_modules", "**/build", "**/.idea", "**/coverage", "**/pnpm-lock.yaml", "**/.changeset"]
+  },
+  "javascript": {
+    "parser": {
+      "unsafeParameterDecoratorsEnabled": true
+    },
+    "formatter": {
+      "semicolons": "asNeeded",
+      "quoteStyle": "single",
+      "trailingCommas": "es5",
+      "lineWidth": 120,
+      "indentStyle": "space"
+    }
+  },
+  "json": {
+    "parser": {
+      "allowComments": true
+    }
+  },
+  "organizeImports": {
+    "enabled": true
+  },
+  "linter": {
+    "enabled": true,
+    "rules": {
+      "recommended": true,
+      "style": {
+        "useImportType": "error",
+        "useNodejsImportProtocol": "off",
+        "noNonNullAssertion": "error"
+      },
+      "correctness": {
+        "noUnusedImports": "error",
+        "noUndeclaredDependencies": "warn",
+        "noInvalidUseBeforeDeclaration": "error"
+      },
+      "suspicious": {
+        "noConfusingVoidType": "off",
+        "noConsole": "error",
+        "noUnsafeDeclarationMerging": "off"
+      },
+      "complexity": {
+        "noBannedTypes": "error"
+      },
+      "nursery": {
+        "noRestrictedImports": {
+          "level": "error",
+          "options": {
+            "paths": {
+              "packages/*": "using imports from packages/* is not allowed"
+            }
+          }
+        }
+      }
+    }
+  },
+  "overrides": [
+    {
+      "include": ["packages/core/**"],
+      "linter": {
+        "rules": {
+          "style": {
+            "noRestrictedGlobals": {
+              "level": "error",
+              "options": {
+                "deniedGlobals": ["Buffer"]
+              }
+            }
+          }
+        }
+      }
+    },
+    {
+      "include": ["jest.config.ts", "./scripts/**"],
+      "linter": {
+        "rules": {
+          "correctness": {
+            "noUndeclaredVariables": "off"
+          }
+        }
+      }
+    },
+    {
+      "include": ["demo/**", "demo-openid/**"],
+      "linter": {
+        "rules": {
+          "suspicious": {
+            "noConsole": "off"
+          }
+        }
+      }
+    },
+    {
+      "include": [
+        "*.test.ts",
+        "**/__tests__/**",
+        "jest.*.ts",
+        "samples/**",
+        "demo/**",
+        "demo-openid/**",
+        "scripts/**",
+        "**/tests/**",
+        "tests/**"
+      ],
+      "linter": {
+        "rules": {
+          "correctness": {
+            "noUndeclaredDependencies": "off"
```

Comment:
Can we enable this for all files with dev dependencies set to false, and then here with dev dependencies true?

https://biomejs.dev/linter/rules/no-undeclared-dependencies/#options

Reply:
This options are available from biome version `2.0.0`

Reply:
Also there is a bug now for `noUndeclaredDependencies` https://github.com/biomejs/biome/issues/4992 which will be fixed in `2.0.0`

Reply:
Ah would be good to use the latest version immediately? Otherwise we have to update again after this pr

Reply:
I think we should wait till 2.0 release https://github.com/biomejs/biome/milestone/6 and comeback again to this

Reply:
Ah didn't see it was not released yet. Makes sense 👍